posthog/batch_exports/service.py

```diff
@@ -70,9 +70,27 @@ class SnowflakeBatchExportInputs:
     role: str | None = None
 
 
+@dataclass
+class PostgresBatchExportInputs:
```

Comment:
Not blocking, can be implemented (or not) later: we're dropping two config options supported by the app:
  - `hasSelfSignedCert`, that we probably want to re-introduce
  - `databaseUrl`, I've seen users try to inject a lot of bad stuff in these (and killing pods on unhandled exceptions from the pg client), that I'm ok with dropping it. We'll want to check how many of these we have on cloud, and whether they can be translated to simple params.

Reply:
If I'm reading the postgres-plugin code correctly:

* `hasSelfSignedCert`: Should be fairly trivial to re-introduce, as `psycopg2.connect` supports an `sslmode` parameter that we can set to `'require'` on `hasSelfSignedCert == "No"`. I'll do that in one of the next commits.
* `databaseUrl`: psycopg2 has a [`parse_dsn` function](https://www.psycopg.org/docs/extensions.html#psycopg2.extensions.parse_dsn) that we can use to parse `databaseUrl` into connection parameters. We can put a `try/except` around that and fail with a non-retryable error. Also okay with doing this in a follow-up commit.

Reply:
> databaseUrl: psycopg2 has a [parse_dsn function](https://www.psycopg.org/docs/extensions.html#psycopg2.extensions.parse_dsn) that we can use to parse databaseUrl into connection parameters

That allows users to pass parameters we'll want control on, like connection timeout, and create fun incidents. I'd take the time to check how many pluginconfigs we'd break if deprecating it (we can still parse_dsn when migrating the config though)

Reply:
Yeah, should be easy to add `parse_dsn` to the migration command when migrating Postgres plugin.

Reply:
I've added support for `hasSelfSignedCert` and ignored `databaseUrl`.

Reply:
I'll be merging this to avoid the PR going stale.

Before migrating any users over we'll check `databaseUrl` usage and see if we need to support more parameters. In the meantime, users can onboard themselves, and the old export app will still be running, so we are not breaking anybody by merging this.